### PR TITLE
[Merged by Bors] - chore(CategoryTheory/Adjunction): move `Adjunction.restrictFullyFaithful` to separate file

### PR DESCRIPTION
**Pull request by dagurtomas**
Also resolves a TODO to add lemmas about `Adjunction.restrictFullyFaithful`

---
- [x] depends on: #12344 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Adjunction/Restrict.lean [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      homEquiv_naturality_left_symm := fun {X' X Y} f g => by
+        apply iD.map_injective
+        simpa [Trans.trans] using (comm1.inv.naturality_assoc f _).symm
+      homEquiv_naturality_right := fun {X Y' Y} f g => by
+        apply iC.map_injective
+        suffices R'.map (iD.map g) ≫ comm2.hom.app Y = comm2.hom.app Y' ≫ iC.map (R.map g) by
+          simp [Trans.trans, this]
+        apply comm2.hom.naturality g }
+#align category_theory.adjunction.restrict_fully_faithful CategoryTheory.Adjunction.restrictFullyFaithful
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Adjunction/Restrict.lean:55]:
```suggestion
#align category_theory.adjunction.restrict_fully_faithful CategoryTheory.Adjunction.restrictFullyFaithful

end CategoryTheory.Adjunction
```

**Comment by joelriou** [comment]
You removed two TODOs from the original file, but I do not see them after your moved the code. There was one about full/faithful separately, and the other one was about lemmas describing the produced adjunction, but I do not see where are these lemmas?!

**Comment by dagurtomas** [comment]
The TODO about full/faithful separately already appeared in the docstring at the beginning of the file, under the title future work: "The statements from Riehl 4.5.13 for adjoints which are either full, or faithful". 

About the lemmas, I added a `simps` attribute which produces lemmas about the unit and counit of the adjuntion. If this is not what was meant by the todo, I can add it back.

**Comment by joelriou** [comment]
The equational lemmas that are automatically produced are not the greatest (because of the presence of `Functor.preimage` in the RHS). For `unit_app`, I would suggest this one instead:
```lean
@[simp, reassoc]
lemma map_restrictFullyFaithful_unit_app (X : C) :
    iC.map ((restrictFullyFaithful iC iD adj comm1 comm2).unit.app X) =
    adj.unit.app (iC.obj X) ≫ R'.map (comm1.hom.app X) ≫ comm2.hom.app (L.obj X) := by
  simp [restrictFullyFaithful]
```
(and something similar for `counit_app`).

**Comment by dagurtomas** [comment]
Ok, thanks for the suggestion, it's done. 

Do I understand correctly that adding specific lemmas for `homEquiv_apply` and `homEquiv_symm_apply` is unnecessary, becuause `homEquiv_unit`/`homEquiv_counit` together with the lemmas I added about unit/counit will simplify expressions of that form automatically?

File: Mathlib/CategoryTheory/Adjunction/Restrict.lean [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    iD.map ((restrictFullyFaithful iC iD adj comm1 comm2).counit.app X) =
+    comm1.inv.app (R.obj X) ≫ L'.map (comm2.inv.app X) ≫ adj.counit.app (iD.obj X) := by
+  simp only [Functor.comp_obj, Functor.id_obj, restrictFullyFaithful, equivOfFullyFaithful,
+    Equiv.instTransSortSortSortEquivEquivEquiv_trans, mkOfHomEquiv_counit_app, Equiv.invFun_as_coe,
+    Equiv.symm_trans_apply, Equiv.symm_symm, Iso.homCongr_symm, Iso.refl_symm, Iso.homCongr_apply,
+    Iso.refl_inv, Iso.symm_hom, Iso.app_inv, id_comp, homEquiv_counit, Functor.map_comp, assoc,
+    Iso.symm_inv, Iso.app_hom, Iso.refl_hom, comp_id, Equiv.coe_fn_symm_mk, Functor.image_preimage,
+    NatIso.cancel_natIso_inv_left]
+  change L'.map (iC.map (𝟙 _)) ≫ _ = _
+  simp
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Adjunction/Restrict.lean:73]:
```suggestion
  dsimp [restrictFullyFaithful]
  simp
```

**Comment by joelriou** [comment]
Thanks!

bors d+

**Comment by dagurtomas** [comment]
bors merge

**Pull request closed without merging**